ci-operator/step-registry/vsphere/disconnecting-network/vsphere-disconnecting-network-commands.sh

```diff
@@ -0,0 +1,130 @@
+        return 1
+    fi
+}
+
+function wait_machineconfig_applied() {
+    local role="${1}" try=0 interval=60
+    num=$(oc get node --no-headers | awk -v var="${role}" '$3 == var' | wc -l)
+    local max_retries; max_retries=$(expr $num \* 10)
+    while (( try < max_retries )); do
+        echo "Checking #${try}"
+        if ! check_latest_machineconfig_applied "${role}"; then
+            sleep ${interval}
+        else
+            break
+        fi
+        (( try += 1 ))
+    done
+    if (( try == max_retries )); then
+        echo >&2 "Timeout waiting for all $role machineconfigs are applied"
+        return 1
+    else
+        echo "All ${role} machineconfigs check PASSED"
+        return 0
+    fi
+}
+
+mirror_registry_url=$(< "${SHARED_DIR}"/mirror_registry_url)
+registry_creds=$(< /var/run/vault/mirror-registry/registry_creds)
+
+#node_name=$(oc get nodes --no-headers | awk '{print $1}' | tail -1)
+
+echo "Updating the global cluster pull secret"
+oc get secret/pull-secret -n openshift-config --template='{{index .data ".dockerconfigjson" | base64decode}}' > /tmp/.dockerconfigjson
+oc registry login --registry="${mirror_registry_url}" --auth-basic="${registry_creds}" --insecure --to=/tmp/.dockerconfigjson
+oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=/tmp/.dockerconfigjson
+#echo "Checking that pull-secret is updated with mirror registry's pull secret"
+#ret=0
+#oc debug node/${node_name} -- -- chroot /host cat /var/lib/kubelet/config.json | grep ${mirror_registry_url} || ret=$?
+#if [[ $ret -ne 0 ]]; then
+#  echo "Could not find mirror registry pull scret in global cluster pull secret"
+#  exit 1
+#fi
+echo "Make sure all machines are applied with latest machineconfig"
+wait_machineconfig_applied "master"
+wait_machineconfig_applied "worker"
+
+echo "Adding the CA signed mirror-registry server cert to cluster"
+client_ca_cert=/var/run/vault/mirror-registry/client_ca.crt
+oc create configmap registry-config --from-file=${mirror_registry_url/:/..}=${client_ca_cert} -n openshift-config
+oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-config"}}}' --type=merge
+#sleep 1min to wait for patch applied
+sleep 60
+#echo "Checking that certs are updated with mirror registry"
+#ret=0
+#oc debug node/${node_name} -- chroot /host ls /etc/docker/certs.d/ | grep "${mirror_registry_url}" || ret=$?
+#if [[ $ret -ne 0 ]]; then
+#  echo "Could not find mirror registry pull scret in global cluster pull secret"
+#  exit 1
+#fi
+
+echo "Adding ICSP to make cluster using the mirror registry"
+oc create -f "${icsp_file}"
+echo "Make sure all machines are applied with latest machineconfig"
+wait_machineconfig_applied "master"
+wait_machineconfig_applied "worker"
+
+echo "Disconnecting network"
+ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no core@"${dns_server}" \
```

Comment:
Though the dns server is the bastion host, maybe getting the bastion ip and user from `${SHARED_DIR}/bastion_private_address` and `${SHARED_DIR}/bastion_ssh_user` is more readable?